### PR TITLE
[components] Switch to mobile on resize. No need to refresh browser

### PR DESCRIPTION
**Pull request by kristofferj**
Switching to responsive when resizing browser. This is nice when developing.
Before it was necessary to refresh the browser.

### CODE REVIEW

**Comment by bjoerge** [comment]
This could need a rebase against current next

**Comment by kristofferj** [comment]
Rebased

**Comment by kristofferj** [comment]
rebased again.

File: packages/@sanity/components/src/panes/SplitController.js [hunk 1]
```diff
@@ -107,7 +132,7 @@ export default class PanesSplitController extends React.Component {
 
     // TODO We need a way to target mobile devices in JS
     // --screen-medium-break: 32em;  ~32 * 16 = 512
-    const isMobile = window && window.innerWidth < 512
+    const isMobile = this.state.windowWidth < 512
```
[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/components/src/panes/SplitController.js]: agreed

File: packages/@sanity/components/src/panes/SplitController.js [hunk 2]
```diff
@@ -18,8 +19,24 @@ export default class PanesSplitController extends React.Component {
     onSholdExpand() {}
   }
 
+  state = {
+    windowWidth: 1000
```
[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/components/src/panes/SplitController.js]: My thoughts was that window is not always available at this point. Maybe i should set it to undefined and set it on componentDidMount?

File: packages/@sanity/components/src/panes/SplitController.js [hunk 3]
```diff
@@ -18,8 +19,24 @@ export default class PanesSplitController extends React.Component {
...
 
+  componentDidMount() {
+    if (window) {
+      window.addEventListener('resize', this.handleResize)
+    }
+  }
+
+  componentWillUnmont() {
+    if (window) {
+      window.removeEventListener('resize', this.handleResize)
```
[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/components/src/panes/SplitController.js]: Does it help if i make it a passive listener?

File: packages/@sanity/components/src/panes/SplitController.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (bjoerge) [reply packages/@sanity/components/src/panes/SplitController.js]: Good point, window is available in componentDidMount. Hang on, I'm pushing a small fix in a moment.

**Comment by bjoerge** [comment]
A change in window width can also be triggered by the `orientationchange` event.

**Comment by bjoerge** [comment]
Did a couple of improvements:
- Now sharing `resize` and `orientationchange` listeners between all SplitPaneController components. This will also synchronize their rendering.
- Debouncing resize/orientationchange events every 50ms

**Pull request merged**